rpc/core/mempool.go

```diff
@@ -42,11 +42,12 @@ func BroadcastTxSync(ctx *rpctypes.Context, tx types.Tx) (*ctypes.ResultBroadcas
 	}
 	r := res.GetCheckTx()
 	return &ctypes.ResultBroadcastTx{
-		Code:      r.Code,
-		Data:      r.Data,
-		Log:       r.Log,
-		Codespace: r.Codespace,
-		Hash:      tx.Hash(),
+		Code:         r.Code,
+		Data:         r.Data,
+		Log:          r.Log,
+		Codespace:    r.Codespace,
+		MempoolError: r.MempoolError,
```

Comment:
What information goes into the `MempoolError` field? In case of exposing server-side error messages that intend to use for causal analysis for a developer or operator, it may give malicious people clues to an attack.

Reply:
`MempoolError` have only `postCheck` error in this changes.

Reply:
Tendermint's `ResultBroadcastTx` does the same thing, setting the `postCheck` error message as the response. However, there is no explicit note for the `postCheck` specification or implementation that "if it returns an error, the message is exposed to the outside world".

I think it's a general security consideration to not allow free text messages and not expose any information beyond "transaction check failed". What do you think?

Reply:
I don't know if an error that occurred on mempool would lead to an attack if it were made public. However, even if the error content is hidden, it seems meaningless because the malicious party can reproduce it at hand.

Reply:
From the looks of it, `postCheck` can be implemented in any way, so errors may unintentionally generate system-related information, such as insufficient memory or storage, missing specified keys, or whether an attack is possible using malleability, etc.

I’ll confirm that it’s assumed with sufficient human checks for the error messages generated by `postCheck`.